src/apps/ociobakelut/main.cpp

```diff
@@ -158,58 +174,111 @@ int main (int argc, const char* argv[])
+                std::cout << *groupTransform;
+                std::cout << "\n";
+            }
+
+            editableConfig->addColorSpace(outputColorSpace);
+            config = editableConfig;
         }
-        if(!shaperspace.empty())
+        else if(groupTransform->getTransform(0)->getTransformType() == OCIO::TransformType::TRANSFORM_TYPE_DISPLAY_VIEW)
         {
-            std::cerr << "\nERROR: --shaperspace is not allowed when using --lut\n\n";
-            std::cerr << "See --help for more info." << std::endl;
-            return 1;
-        }
+            if(inputspace.empty())
+            {
+                std::cerr << "\nERROR: --inputspace is required when using --view\n\n";
+                std::cerr << "See --help for more info." << std::endl;
+                return 1;
+            }
+            if(!outputspace.empty())
+            {
+                std::cerr << "\nERROR: --outputspace is not allowed when using --view\n\n";
+                std::cerr << "See --help for more info." << std::endl;
+                return 1;
+            }
+            if(view.empty())
+            {
+                std::cerr << "\nERROR: Specifying the view is required when using --view\n\n";
+                std::cerr << "See --help for more info." << std::endl;
+                return 1;
+            }
+            if(display.empty())
+            {
+                std::cerr << "\nERROR: Specifying the display is required when using --view\n\n";
+                std::cerr << "See --help for more info." << std::endl;
+                return 1;
+            }
 
-        OCIO::ConfigRcPtr editableConfig = OCIO::Config::Create();
+            // Creating a new config
+            OCIO::ConfigRcPtr editableConfig = OCIO::Config::Create();
 
-        OCIO::ColorSpaceRcPtr inputColorSpace = OCIO::ColorSpace::Create();
-        inputspace = "RawInput";
-        inputColorSpace->setName(inputspace.c_str());
-        editableConfig->addColorSpace(inputColorSpace);
+            //Setting up input ColorSpace
+            OCIO::ColorSpaceRcPtr inputColorSpace = OCIO::ColorSpace::Create();
+            inputColorSpace->setName(inputspace.c_str());
+            editableConfig->addColorSpace(inputColorSpace);
 
-        OCIO::ColorSpaceRcPtr outputColorSpace = OCIO::ColorSpace::Create();
-        outputspace = "ProcessedOutput";
-        outputColorSpace->setName(outputspace.c_str());
+            //Setting up the output ColorSpace
+            OCIO::ColorSpaceRcPtr outputColorSpace = OCIO::ColorSpace::Create();
+            outputspace = "ProcessedOutput";
+            outputColorSpace->setName(outputspace.c_str());
 
-        outputColorSpace->setTransform(groupTransform,
-            OCIO::COLORSPACE_DIR_FROM_REFERENCE);
+            // Creating the transform and applying to outputColorSpace
+            OCIO::DisplayViewTransformRcPtr t = OCIO::DisplayViewTransform::Create();
```

Comment:
This transform already exists in the group transform, albeit without a src color space. Recommend refactoring to prevent defining it twice. In general I think the `--display` option can be treated more like `--outputspace` than `--lut` in terms of matching implementation behavior. A display/view pair is in essence just another way of defining the output color space. It's more than that under the hood, but in terms of user experience it's equivalent.